src/AstTranslator.cpp

```diff
@@ -1052,9 +1080,7 @@ std::unique_ptr<RamStatement> AstTranslator::translateRecursiveRelation(
 
         /* create update statements for fixpoint (even iteration) */
         appendStmt(updateRelTable,
-                std::make_unique<RamSequence>(
-                        std::make_unique<RamMerge>(std::unique_ptr<RamRelationReference>(rrel[rel]->clone()),
-                                std::unique_ptr<RamRelationReference>(relNew[rel]->clone())),
+                std::make_unique<RamSequence>(genMerge(rrel[rel].get(), relNew[rel].get()),
```

Comment:
Yes, that is correct. They are prototypical for generating the relation references for the relation/delta/new versions of a relation. 

This needs to be tidied up. The AstTranslator has the issue that the Relation processing accumulated quite a bit of dust and overly complicated.

Reply:
Ah, they do get cloned in genMerge. I missed the called to genMerge, so it looked like the raw pointers ended up in two unique pointers - but they don't.

It might be safer to pass unique pointer references to genMerge rather than raw pointers.